pkg/proc/arm64_arch.go

```diff
@@ -406,3 +409,45 @@ func (a *ARM64) AddrAndStackRegsToDwarfRegisters(staticBase, pc, sp, bp, lr uint
 		LRRegNum:   arm64DwarfLRRegNum,
 	}
 }
+
+func (a *ARM64) DwarfRegisterToString(name string, reg *op.DwarfRegister) string {
```

Comment:
Unless I'm missing something `a` (method receiver) isn't actually being used here. What if this (and the other arch implementation) is a `String` method on the register type?

Reply:
It doesn't use a but it's different depending on the architecture. One thing we could do is stuff a function pointer inside op.DwarfRegister and then its string method would be:

```
func (reg *DwarfRegister) String() string {
    return reg.fmtFunc(reg)
}
```

However at some point we'll also want a way to get the register name given a register number and DwarfRegisterToString could be extended to do that, otherwise we'll need something else (probably another method of Arch).

Reply:
Correction: it would be proc.Register not op.DwarfRegister to get the String method, of course.

Reply:
I was thinking more each concrete implementation would have this method, but after thinking it through a bit more I think your approach is cleaner.

Reply:
I think it would be a shame to make it into an interface just for this thing, given that everything else is the same. Also we'll have to go back from register numbers to this interface objects somehow, and I don't know how good that would look.